### ISSUE TITLE
postcss-reporter in gulp reports all stylelint errors as warnings

### ISSUE THREAD

**Issue by onigoetz** [stylelint/stylelint#842]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

When I use `stylelint` with `postcss` inside `gulp`, the error reporting marks everything as warning.

> Which rule, if any, is this issue related to?

none specifically

> What CSS is needed to reproduce this issue?

Any CSS that triggers a warning

> What stylelint configuration is needed to reproduce this issue?

Any rule that throws an error

> Which version of stylelint are you using?

`4.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

postcss, but the CLI does the same, but in the CLI the `noIcon` option is enabled so it's not visible

> Does your issue relate to non-standard syntax ?

no, only to the reporter

> What did you expect to happen?

The two first elements should have been warnings, and the rest should be errors

```shell
css/components/_Panel.scss
7:9     ⚠  'margin-bottom' is not recommended, read more here : https://... [stylelint]
12:9    ⚠  'margin-bottom' is not recommended, read more here : https://... [stylelint]
29:13   Nesting exceeds maximum nesting depth (statement-max-nesting-depth) [stylelint]
29:14   Expected class selector ".expanded" to match specified pattern (selector-class-pattern) [stylelint]
```

> What actually happened ?

The following warnings were flagged:

```shell
css/components/_Panel.scss
7:9     ⚠  'margin-bottom' is not recommended, read more here : https://... [stylelint]
12:9    ⚠  'margin-bottom' is not recommended, read more here : https://... [stylelint]
29:13   ⚠  Nesting exceeds maximum nesting depth (statement-max-nesting-depth) [stylelint]
29:14   ⚠  Expected class selector ".expanded" to match specified pattern (selector-class-pattern) [stylelint]
```

**Comment by onigoetz** [comment]
I made some investigation, and it seems that `postcss-reporter` expects the key for the severity to be `type: error` but `stylelint` sets a `severity: error` instead

**Comment by davidtheclark** [comment]
All of stylelint's complaints are registered with PostCSS as PostCSS Warning objects, which do not have severity levels. It's only the stylelint-specific reporters (what you get with the CLI and Node API) that differentiate between severity levels. postcss-reporter treats all PostCSS warnings as "warnings". (PostCSS also throws CSS syntax errors, and there's are the "errors".)

postcss-reporter needs to stay as generic as it is rather than looking specifically for stylelint data (e.g. severity) on the warning.

So at this point the only way I can think of to improve the experience for people who want to use stylelint with gulp is to write a gulp-stylelint plugin. I will make a separate issue for that.

**Comment by jeddy3** [comment]
> So at this point the only way I can think of to improve the experience for people who want to use stylelint with gulp is to write a gulp-stylelint plugin. I will make a separate issue for that.

https://www.npmjs.com/package/gulp-stylelint :)

**Comment by davidtheclark** [comment]
Ok -- ha! I was just writing an issue for creating gulp-stylelint: I forgot about that. I think that maybe we should push that as the best way to use stylelint with gulp, rather than including it in the chain of PostCSS plugins. As part of the chain it is limited in how it can report; as its own plugin, it could do everything the CLI and Node API can do.

So @onigoetz my recommendation to you is that you try gulp-stylelint.

**Comment by jeddy3** [comment]
> I think that maybe we should push that as the best way to use stylelint with gulp, rather than including it in the chain of PostCSS plugins

Agreed. I'll make separate issue for that.

**Comment by davidtheclark** [comment]
Ok. I'll close this, then, because it's not really an issue with stylelint.

**Comment by onigoetz** [comment]
Actually, there is something that can be changed in stylelint.

as for the moment, the information if it's a warning or error are already provided and `postcss-reporter` even if it is not standard, does some recognition to see if it is a warning or not.

But for the moment the parameter name is `severity` in `stylelint` and `type` in `postcss-reporter`.
